src/driver.ts

```diff
@@ -534,10 +615,25 @@ export abstract class CMakeDriver implements vscode.Disposable {
       flags.push(...util.objectPairs(this._kit.cmakeSettings).map(([key, val]) => _makeFlag(key, util.cmakeify(val))));
     }
 
+    // Expand all flags
     const final_flags = flags.concat(settings_flags);
-    log.trace('CMake flags are', JSON.stringify(final_flags));
+    const expanded_flags_promises = final_flags.map(
+        async (value: string) => this.expandString(value, await this.getExpandedConfigureEnvironment()));
+    const expanded_flags = await Promise.all(expanded_flags_promises);
+    log.trace('CMake flags are', JSON.stringify(expanded_flags));
+
+    // Setup temporary environment for the configure step
+    const old_env = process.env;
+    const configure_env = util.mergeEnvironment(old_env as proc.EnvironmentVariables,
+                                                await this.getExpandedEnvironment(),
+                                                await this.getExpandedConfigureEnvironment());
+    process.env = configure_env;
+
+    const retc = await this.doConfigure(expanded_flags, consumer);
```

Comment:
The setting the environment then resetting is spooky to me. In this case in particular, because if `doConfigure` throws we won't reset the environment.

Reply:
The only reason I am setting it here is because of the legacy driver. The environment for the cmake-server gets set when starting it. I could add an (optional) environment argument to the `doConfigure()` though.